s3-connector/unittest_s3-connector.py

```diff
@@ -0,0 +1,81 @@
+import boto3
+import unittest
+import s3connector
+import os
+import logging
+import yaml
+from yaml.loader import SafeLoader
+
+
+class Tests3Bucket(unittest.TestCase):
+    @classmethod
+    def setUpClass(self):
+        cfg_path = os.path.dirname(os.path.abspath(__file__))
+        cfg_file = cfg_path + "/" + "test_config.yaml"
+        self.test_cfg = {}
+        with open(cfg_file) as c_info_file:
+             self.test_cfg = yaml.load(c_info_file, Loader=SafeLoader)
+        self.test_cfg = self.test_cfg["test_arguments"]
+        self.key = self.test_cfg['key']
+        self.secret= self.test_cfg['secret']
+        self.s3_endpoint=self.test_cfg['s3_endpoint']
+        self.region_name=self.test_cfg['region_name']
+        # Intialize connection for S3
+        self.s3_connection = s3connector.S3(
+            self.test_cfg["key"], 
+            self.test_cfg["secret"], 
+            self.test_cfg["s3_endpoint"], 
+            self.test_cfg["region_name"]
+            )
+
+    # Test connection for S3
+    def test_connection(self):
+        self.assertTrue(self.s3_connection)
+
+    # Test list buckets from s3 
+    def test_list_bucket(self):
+        conn = self.s3_connection.list_buckets()
+        for bucket in conn:
+            print(bucket.name)
```

Comment:
Here need to check if the expected bucket names are listed? Hence you should know in advance what are the buckets and make sure they are listed!

Reply:
@orionsBeltWest I have added function to test the connection with wrong credentials. Please review and let me know.